Cron/ReconcileLostPayments.php

```diff
@@ -0,0 +1,237 @@
+
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * This class attempts to reconcile payments that have dropped off early e.g. they
+ * fell out of the flow after draw-down but before redirecting back to the merchant.
+ *
+ * It also prolongs iwocaPay orders so that they aren't automatically cleared after
+ * a user set timeout. Instead, 48 hours is used.
+ */
+class ReconcileLostPayments
+{
+    /**
+     * @var OrderFactory
+     */
+    protected $orderFactory;
+    /**
+     * @var IwocaClientFactory
+     */
+    protected $iwocaClientFactory;
+    /**
+     * @var Config
+     */
+    private $config;
+    /**
+     * @var Json
+     */
+    private $jsonSerializer;
+    /**
+     * @var CollectionFactory
+     */
+    protected $orderCollectionFactory;
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var TimezoneInterface
+     */
+    protected $timezone;
+
+    /**
+     * @var ResourceConnection
+     */
+    protected $resourceConnection;
+
+    public function __construct(
+        CollectionFactory  $orderCollectionFactory,
+        LoggerInterface    $logger,
+        OrderFactory       $orderFactory,
+        IwocaClientFactory $iwocaClientFactory,
+        Config             $config,
+        Json               $jsonSerializer,
+        TimezoneInterface  $timezone,
+        ResourceConnection $resourceConnection
+    )
+    {
+        $this->orderCollectionFactory = $orderCollectionFactory;
+        $this->logger = $logger;
+        $this->orderFactory = $orderFactory;
+        $this->iwocaClientFactory = $iwocaClientFactory;
+        $this->config = $config;
+        $this->jsonSerializer = $jsonSerializer;
+        $this->timezone = $timezone;
+        $this->resourceConnection = $resourceConnection;
+    }
+
+    /**
+     * Returns all orders which were completed using iwocaPay and are in the "pending payment" state
+     *
+     * @return array
+     */
+    protected function getPendingIwocaPayOrders(): array
+    {
+        $orders = [];
+        foreach ($this->orderFactory->create()->getCollection() as $order) {
+            if ($order->getPayment()->getMethod() === 'iwocapay' && $order->getStatus() === 'pending_payment') {
+                $customerId = $order->getCustomerId();
+                $orders[$customerId][] = $order;
+            }
+        }
+        return $orders;
+    }
+
+    /**
+     * Returns an order_id for an order. This is parsed from the comment history as there is no place to store
+     * it within Magento. It may be null if no order_id is found.
+     *
+     * @param $order
+     * @return string|null
+     */
+    protected function findUUIDInOrderComments($order): ?string
+    {
+        foreach ($order->getStatusHistoryCollection() as $comment) {
+            if (preg_match('/"\b[A-Fa-f0-9]{8}-[A-Fa-f0-9]{4}-4[A-Fa-f0-9]{3}-[89ABab][A-Fa-f0-9]{3}-[A-Fa-f0-9]{12}\b"/', $comment->getComment(), $matches)) {
```

Comment:
I like this idea, I think we should invest in this in the future, for now I'm gonna stick with so that we can get this out. I'll create a ticket for someone to go back and add it in - could be a good intro into Magento for someone else